addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js

```diff
@@ -270,6 +271,12 @@ export class AnalyticDistribution extends Component {
         const values = {};
         // Analytic Account fields
         line.analyticAccounts.map((account) => {
+            let domain = [["root_plan_id", "=", account.planId]];
+            if ("company_id" in this.props.record.activeFields) {
+                domain = Domain.and([domain, [
+                    ["company_id", "in", [this.props.record.data.company_id[0], false]],
```

Comment:
There are 2 issues here:

1. There is a stack trace when the company is not set
2. Parent analytic accounts are excluded